packages/components/src/internal/components/ontology/actions.ts

```diff
@@ -166,8 +170,31 @@ export function fetchParentPaths(conceptPath: string): Promise<PathModel[]> {
     return getConceptParentPaths(conceptPath);
 }
 
-export function fetchConceptForCode(code: string): Promise<ConceptModel> {
-    return Ontology.getConcept(code);
+/**
+ * Concatenate the parent concept codes minus the root
+ * @param parents
+ */
+export function getParentsConceptCodePath(parents: PathModel[]): string {
+    return [...parents]
+        .filter(node => !node.code.startsWith(ONTOLOGY_ROOT_CODE_PREFIX)) // Ignore the root node
+        .map(node => node.code)
+        .join('/');
+}
+
+export const CONCEPT_CACHE = new Map<string, ConceptModel>();
+export async function fetchConceptForCode(code: string): Promise<ConceptModel> {
+    if (!CONCEPT_CACHE.has(code)) {
+        const concept = await Ontology.getConcept(code);
+        CONCEPT_CACHE.set(code, concept);
+    }
+
+    return CONCEPT_CACHE.get(code);
+}
+
+export function getConceptForCode(code: string): ConceptModel {
+    if (!code) return undefined;
+    fetchConceptForCode(code);
+    return CONCEPT_CACHE.get(code);
```

Comment:
Is there a race condition here? Between `fetchConceptForCode` setting the cache and the value being returned.

Reply:
Yes, potentially there is. This however was my compromise to the use case for it, which is in the domain designer field row details section display for the selected ontology concept. This `getDetailsTextArray` function really doesn't want to be async and since it is called several times during the loading of the domain field information, the concept info will be loaded after the first time. If this becomes an issue in the display, we can look to "fix it", but I wasn't able to repro a case where it didn't show the expected value.